source/Ship.cpp

```diff
@@ -1302,6 +1302,31 @@ void Ship::Place(Point position, Point velocity, Angle angle, bool isDeparting)
 				bay.ship->SetSwizzle(bay.ship->customSwizzle >= 0 ? bay.ship->customSwizzle : swizzle);
 		}
 	}
+
+	// Get the weapon ranges for this ship, so the Ai can call it.
+	for(const auto &hardpoint : armament.Get())
+	{
+		const Weapon *weapon = hardpoint.GetOutfit();
+		if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoint.IsTurret())
+			turretRange = max(turretRange, weapon->Range() + hardpoint.GetPoint().Length());
+		else if(weapon && !weapon->Ammo() && weapon->DoesDamage() && hardpoint.IsGun())
+			gunRange = max(gunRange, weapon->Range() + hardpoint.GetPoint().Length());
```

Comment:
```suggestion
		const Weapon *weapon = hardpoint.GetOutfit();
		if(!weapon || !weapon->Ammo() || !weapon->DoesDamage())
			continue;
		double range = weapon->Range() + hardpoint.GetPoint().Length();
		if(hardpoint.IsTurret())
			turretRange = max(turretRange, range);
		else if(hardpoint.IsGun())
			gunRange = max(gunRange, range);
```
just for optimisation, should work the same way

Reply:
we need to recalculate this whenever weapons are changed btw so this should prob be moved
but first it should be changed to this